src/main/java/lv/ctco/javaschool/app/boundary/TripApi.java

```diff
@@ -48,4 +66,60 @@ public ListTripDto getActiveTrips() {
                 .collect(Collectors.toList()));
         return listTripDto;
     }
+
+    @GET
+    @Path("/places")
+    @Produces("application/json")
+    @RolesAllowed({"ADMIN", "USER"})
+    public List<Place> getAllPlaces() {
+        return Arrays.asList(Place.values());
+    }
+
+    @POST
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/createTrip")
+    public void createNewTrip(JsonObject field) {
+        User user = userStore.getCurrentUser();
+        Trip trip = new Trip();
+        trip.setDriver(user);
+        for (Map.Entry<String, JsonValue> pair : field.entrySet()) {
+            String addr = pair.getKey();
+            String value = ((JsonString) pair.getValue()).getString();
+            trip = setFieldsToTrip(trip, addr, value);
+        }
+        tripStore.addTrip(trip);
+    }
+
+    private Trip setFieldsToTrip(Trip trip, String addr, String value) throws IllegalArgumentException {
```

Comment:
`IllegalArgumentException` it's an unchecked exception, so don't need to declare it in a throws clause.